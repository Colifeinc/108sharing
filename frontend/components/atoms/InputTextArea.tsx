import * as React from 'react'
import FormItem from 'components/layouts/FormItem'

interface IProps {
  name: string
  defaultValue: string
  label?: null | string
  placeholder?: null | string
  error?: null | string
  required?: boolean
}

const InputTextArea = React.forwardRef<HTMLTextAreaElement, IProps>((props, ref) => {
  const { name, defaultValue, label, placeholder, error, required } = props

  return (
    <FormItem label={label} required={required} error={error}>
      <textarea
        className="
          block w-full h-40 px-3 py-2 bg-neutral-100 rounded outline-none
          border-2 border-transparent
          focus:border-2 focus:border-primary focus:bg-white
        "
        id={name}
        name={name}
        placeholder={placeholder}
        required={required}
        defaultValue={defaultValue}
        ref={ref}
      />
    </FormItem>
  )
})

export default InputTextArea
