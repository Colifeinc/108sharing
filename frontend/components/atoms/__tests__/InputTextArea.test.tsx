// tslint:disable-next-line
import '../../../test/matchMedia.mock'

import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import * as React from 'react'
import InputTextArea from '../InputTextArea'

jest.mock('../../../core/i18n')

describe('atoms/InputTextArea.tsx', () => {
  const props = {
    name: 'test',
    defaultValue: 'test',
  }

  it('should matches to the snapshot', () => {
    const wrapper = shallow(<InputTextArea {...props} />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })

  // it('should matches to the snapshot with label', () => {
  //   const wrapper = shallow(<InputTextArea {...props} label="Label" />)
  //   expect(wrapper.find('.InputTextArea_Title').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })

  // it('should matches to the snapshot with required', () => {
  //   const wrapper = shallow(<InputTextArea {...props} label="Label" required={true} />)
  //   expect(wrapper.find('.InputTextArea_Required').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })

  // it('should matches to the snapshot with error', () => {
  //   const wrapper = shallow(<InputTextArea {...props} error={'error'} />)
  //   expect(wrapper.find('.InputTextArea_Error').length).toBe(1)
  //   expect(toJson(wrapper)).toMatchSnapshot()
  // })
})
