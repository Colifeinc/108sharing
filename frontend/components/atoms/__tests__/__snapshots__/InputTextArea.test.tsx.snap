// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`atoms/InputTextArea.tsx should matches to the snapshot 1`] = `
<FormItem>
  <textarea
    className="
          block w-full h-40 px-3 py-2 bg-neutral-100 rounded outline-none
          border-2 border-transparent
          focus:border-2 focus:border-primary focus:bg-white
        "
    defaultValue="test"
    id="test"
    name="test"
  />
</FormItem>
`;
